backend/src/nodes/image_util_nodes.py

```diff
@@ -554,3 +554,37 @@ def run(
         ssim = round(calculate_ssim(comp_img, orig_img), 6)
 
         return (float(mse), float(psnr), ssim)
+
+@NodeFactory.register("chainner:image:canny_edge_detection")
+class canny_edge_detection(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Detect edges and output as an image"
+        self.inputs = [
+            ImageInput(),
+            SliderInput("Lower Threshold", maximum=200, default=50, precision=1),
+            SliderInput("Higher Threshold", maximum=200, default=150, precision=1),
+            ]
+        self.outputs = [ImageOutput(image_type="Input0")]
+        self.category = ImageUtilityCategory
+        self.name = "Canny Edge Detection"
+        self.icon = "MdAutoFixHigh"
+        self.sub = "Miscellaneous"
+
+    def run(
+        self,
+        img: np.ndarray,
+        t_lower: int,
+        t_upper: int,
+    ) -> np.ndarray:
+        
+        def unnormalize(img: np.ndarray) -> np.ndarray:
+            img *= 255/img.max()
```

Comment:
Why do you divide 255 by img.max here?

Reply:
Yeah, this is not correct. Dividing by img.max() sets the maximum pixel value to 1 and scale all the pixel values, shifting the image color. This should be `img = (img * 255).astype(np.uint8)`, as cv2.Canny simply requires the image to be in 8-bit color. I would just write this as a single line of code as well and skip the overhead on a method call.